devito/passes/iet/definitions.py

```diff
@@ -340,9 +374,22 @@ def place_definitions(self, iet, globs=None, **kwargs):
         for i in FindSymbols().visit(iet):
             if i in defines:
                 continue
+
             elif i.is_LocalObject:
                 self._alloc_object_on_low_lat_mem(iet, i, storage)
-            elif i.is_Array or i.is_Bundle:
+
+            elif i.is_Bundle:
```

Comment:
Is there any way to reduce the "death by conditionals" here?

Reply:
After spending a couple of hours on this very matter this morning, I'm inclined to say that it's pretty challenging to come up with one. The problem is that there's a combinatorial explosion of possibilities (luckily with small exponent), and all those `if`s implement such space

Reply:
I suppose it's all attributes too, which makes it even more challenging without obfuscating what's going on

Reply:
if the implicit question was "can you use singledispatch", yeah, you got it right